### PR TITLE
Move general-purpose helpers to nyxt/utilites.

### LINKED ISSUE #2399: Move misc utilities to their own files?
It'd be useful to move `funcall*`, `trim-list`, `last-word`, etc. (most are in
`configuration.lisp`) to their own file.

I'd even say their own package, which is then `cl:use`d in `nyxt` and
`nyxt-user`.  This way we can easily list them all by inspecting the package or
through completion.

What shall we call this file?  `utilities` is one that comes to mind, but the
name is rather meaningless.

Thoughts?

### PR DESCRIPTION
**Pull request by Ambrevar**
# Description

Move general purpose helpers to their own package.
This makes it easier to list them (e.g. via package description or via completion).

Note: I've remove the unused (and funky) `trim-list`.

Fixes #2399

# Discussion

Did I miss some utility functions?

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.  

- [x] I have pulled from master before submitting this PR
- [x] My code follows the style guidelines for Common Lisp code
  - See [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf) and [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml).
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer (the peer review to approve a PR counts.  The reviewer must download and test the code)
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

File: source/utilities.lisp [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+  (open pathname :direction :probe :if-does-not-exist :create))
+
+(export-always 'funcall*)
+(defun funcall* (f &rest args)
+  "Like `funcall' but does nothing when F is nil."
+  (when f (apply #'funcall f args)))
+
+(defun guess-external-format (filename)
+  (or (swank-backend:guess-external-format filename)
+      (swank-backend:find-external-format "latin-1")
```
Comment (aartaka) [inline source/utilities.lisp:60]: Why not UTF-8?

File: source/utilities.lisp [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+      ""
+      (the (values string &optional) (alex:last-elt (sera:words s)))))
+
+(export-always 'make-ring)
+(defun make-ring (&key (size 1000))
+  "Return a new ring buffer."
+  (containers:make-ring-buffer size :last-in-first-out))
+
+(export-always 'on)
+(defmacro on (hook args &body body)
```
Comment (aartaka) [inline source/utilities.lisp:117]: Note that [I've recently added `on` and `once-on` to `nhooks`](https://github.com/atlas-engineer/nhooks/commit/7be7c652e66abfc4b3d482a78ae4b7bd32f03118), so those can be removed.

**Review by aartaka — APPROVED** [review]

File: source/utilities.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/utilities.lisp:60]: That's how swank does it and what seems to work.  No idea why O.o

File: source/utilities.lisp (hunk 2 above)
Comment (Ambrevar) [reply source/utilities.lisp:117]: Thanks for the update!

File: source/utilities.lisp (hunk 1 above)
Comment (Ambrevar) [reply source/utilities.lisp:60]: To clarify, it does not do the right thing when I used UTF-8, unless I did something else wrong.

**Pull request closed without merging**